sbol_utilities/component.py

```diff
@@ -624,7 +624,69 @@ def part_in_backbone(identity: str, part: sbol3.Component, backbone: sbol3.Compo
     part_in_backbone_component.types.append(topology_type)
     return part_in_backbone_component, part_in_backbone_seq
 
-def digestion(reactant:sbol3.Component, restriction_enzymes:List[sbol3.ExternallyDefined], assembly_plan:sbol3.Component)-> Tuple[sbol3.Component, sbol3.Sequence]:
+def part_in_backbone_from_sbol(identity: Union[str, None],  sbol3_comp: sbol3.Component, part_location: List[int], part_roles:List[str], fusion_site_length:int, linear:bool=False, **kwargs) -> Tuple[sbol3.Component, sbol3.Sequence]:
+    """Creates a Part in Backbone Component and its Sequence following BP011 from an unformatted SBOL3 Component.
+    It overwrites the SBOL3 Component provided. 
+    A part inserted into a backbone is represented by a Component that includes both the part insert 
+    as a feature that is a SubComponent and the backbone as another SubComponent.
+    For more information about BP011 visit https://github.com/SynBioDex/SBOL-examples/tree/main/SBOL/best-practices/BP011 
+
+    :param identity: The identity of the Component, is its a String it build a new SBOL Component, if None it adds on top of the input. The identity of Sequence is also identity with the suffix '_seq'.
+    :param sbol3_comp: The SBOL3 Component that will be used to create the part in backbone Component and Sequence.
+    :param part_location: List of 2 integers that indicates the start and the end of the unitary part. Note that the index of the first location is 1, as is typical practice in biology, rather than 0, as is typical practice in computer science.
+    :param part_roles: List of strings that indicates the roles to add on the part.
+    :param fusion_site_length: Integer of the length of the fusion sites (eg. BsaI fusion site lenght is 4, SapI fusion site lenght is 3)
+    :param linear: Boolean than indicates if the backbone is linear, by default it is seted to Flase which means that it has a circular topology.    
```

Comment:
```suggestion
    :param linear: Boolean that indicates if the backbone is linear, by default it is set to False, which means it has a circular topology.    
```
For a TODO, note that we should probably be able to infer this from the input, rather than having to supply it explicitly.

Reply:
added